crypto/bn/bn_nist.c

```diff
@@ -319,6 +319,35 @@ static void nist_cp_bn(BN_ULONG *dst, const BN_ULONG *src, int top)
 # endif
 #endif                          /* BN_BITS2 != 64 */
 
+#ifdef NIST_INT64
+# if BN_BITS2 == 64
+/* Helpers to load/store a 32-bit word (unsigned int) from/into a memory
+ * location and avoid potential aliasing issue.  */
+static ossl_inline unsigned int load_u32(const void *ptr)
+{
+    unsigned int tmp;
```

Comment:
Why not:

(1) use `uint32_t` here as the type;
(2) and always use this definition? This is the idiomatic way to do this, and compilers will optimise to a single load instruction on systems where unaligned access is permitted

Reply:
For (1) I mistakenly assume I couldn't use uint32_t because OpenSSL is C89 compatible (I didn't see the customized uint32_t typedef in e_os2.h).

For (2) because the store_u32 implementation is not compatible with 32-bit big-endian targets where long long is available.  I can add a cast into store_u32 to solve this issue though.

I've made the changes following your advice and now I'm testing it.  Will force push after the test succeeds.

Reply:
Done.